src/main/java/pl/asie/ynot/oc/OCChannelType.java

```diff
@@ -25,8 +35,17 @@ public String getName() {
 
     @Override
     public boolean supportsBlock(@Nonnull World world, @Nonnull BlockPos pos, @Nullable EnumFacing side) {
-        return world.getTileEntity(pos.offset(side)) instanceof Environment
-                || world.getTileEntity(pos.offset(side)) instanceof SidedEnvironment;
+        TileEntity tile = world.getTileEntity(pos.offset(side));
+
+        if (tile != null) {
+            if(tile.hasCapability(SIDED_ENVIRONMENT_CAPABILITY, side)) {
+                return true;
+            } else if(tile.hasCapability(ENVIRONMENT_CAPABILITY, side)) {
+                return true;
+            }
+        }
```

Comment:
If you're supporting capabilities exclusively, you may want to correct the registration code in the YNot class to not register an IConnectable for instanceof class checks.

Reply:
Addressed in deee237